symplyphysics/laws/kinematics/displacement_in_simple_harmonic_motion.py

```diff
@@ -7,80 +7,62 @@
 function of time.
 """
 
-from sympy import Eq, cos, symbols, dsolve, Function as SymFunction
+from sympy import Eq, cos, dsolve
 from symplyphysics import (
-    units,
     Quantity,
-    Symbol,
-    angle_type,
     validate_input,
+    symbols,
+    clone_as_symbol,
+    clone_as_function,
 )
 from symplyphysics.core.expr_comparisons import expr_equals
 from symplyphysics.core.quantity_decorator import validate_output_same
 from symplyphysics.core.symbols.quantities import scale_factor
 from symplyphysics.definitions import harmonic_oscillator_is_second_derivative_equation as harmonic_eqn
 
-displacement = symbols("displacement", real=True)
+displacement = clone_as_symbol(
+    symbols.any_dimension,
+    display_symbol="q",
+    display_latex="q", 
+)
 """
-Displacement from rest, usually a function of time.
-
-Symbol:
-    :code:`q`
+Displacement from rest, usually a function of :symbols:`time`. See :symbols:`any_dimension`.
 """
 
-amplitude = symbols("amplitude", positive=True)
-r"""
-The maximum absolute value of the displacement.
-
-Symbol:
-    :code:`q_max`
-
-Latex:
-    :math:`q_\text{max}`
+amplitude = clone_as_symbol(
```

Comment:
I guess I would for now refrain from making it a separate operation. It's actually not just `max(q)` but $`\max_{t \in \mathbb{R}}{|q(t)|}`$ since it's the amplitude and not simply the maximum value of the function.